packages/deployer/test/src/tasks/deploy.test.js

```diff
@@ -1,13 +1,18 @@
 const path = require('path');
 const { rejects } = require('assert/strict');
 const { copyFile, unlink, readFile, writeFile } = require('fs/promises');
+const sampleProjectInitializer = require('../../fixture-projects/sample-project/test/helpers/initializer');
 const { ContractValidationError } = require('../../../internal/errors');
 const { useEnvironment } = require('../../helpers');
 
 describe('tasks/deploy.js', function () {
   describe('when deploying a correctly written project', function () {
     useEnvironment('sample-project');
 
+    beforeEach('initialize sample-project owner', async function () {
+      await sampleProjectInitializer(this.deploymentInfo, this.hre);
+    });
```

Comment:
I removed all the Initializer logic from the helper, and let the specific test take care of this because we were only needing it here. And doing it this way we remove a lot of logic from the helper.